src/huggingface_hub/constants.py

```diff
@@ -16,10 +16,18 @@
 
 HUGGINGFACE_CO_URL_HOME = "https://huggingface.co/"
 
-HUGGINGFACE_CO_URL_TEMPLATE = (
-    "https://huggingface.co/{repo_id}/resolve/{revision}/{filename}"
+ENV_VARS_TRUE_VALUES = {"1", "ON", "YES", "TRUE"}
+_staging_mode = (
+    os.environ.get("HUGGINGFACE_CO_STAGING", "NO").upper() in ENV_VARS_TRUE_VALUES
+)
+
+ENDPOINT = (
+    "https://moon-staging.huggingface.co" if _staging_mode else "https://huggingface.co"
 )
```

Comment:
the CI env var itself could be the endpoint `https://moon-staging.huggingface.co` which would be neat b/c then this doesn't appear in the code